### PR TITLE
Build budget app

### PR DESCRIPTION
**Pull request by Outkarm**
Pull Request Description:

This pull request introduces the implementation of the Ruby on Rails capstone project. The project is a mobile web application for managing budgets, allowing users to track their transactions and expenses by category.

Changes Made:
- Set up the repository and project tools.
- Created models based on the ERD diagram, including validations and tests.
- Implemented authentication and authorization using Devise.
- Created the category page where users can view and manage their categories, including the total amount of transactions for each category.
- Implemented the transactions page for each category, displaying a list of transactions ordered by the most recent, along with the total amount for the category.
- Added functionality to add new categories and transactions through dedicated form pages.
- Ensured adequate testing coverage for the application.
- Deployed the project online.
- Recorded a video demonstrating the project features, including a description of the project, a demo of the functionalities, and highlighting interesting pieces of code.

Additional Changes:
- Followed the design guidelines provided, including colors, typography, and layout.
- Used Postgres as the database for the project.
- Implemented input validation to enhance security and prevent malicious actions.
- Chose a view template engine (.erb, .slim, .haml) for rendering views.

Please review the changes and provide feedback.

### CODE REVIEW

**Comment by Outkarm** [comment]
please in case of any change, you can reach out to me on Slack or Zoom, I am available.
it is almost the deadline so I am on edge

File: README.md [hunk 1]
```diff
@@ -1,24 +1,216 @@
...
+
+## ⭐️ Show your support <a name="support"></a>
+
+If you like this project then please do follow me and give a star to this repo.
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- ACKNOWLEDGEMENTS -->
+
+## 🙏 Acknowledgments <a name="acknowledgements"></a>
```
Comment (t-yanick) [inline README.md:190]: - [ ] Great job working on this capstone. As per project requirements, you are supposed to give appropriate credit to [Gregoire Vella on Behance](https://www.behance.net/gregoirevella) the [author of the original design](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=), as required by the [Creative Commons license of the design](https://creativecommons.org/licenses/by-nc/4.0/). Kindly do the Creative commons acknowledgement.

File: README.md [hunk 2]
```diff
@@ -1,24 +1,216 @@
...
+
+  - Yes, It will be my pleasure to help you to solve your problem in your code.
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- LICENSE -->
+
+## 📝 License <a name="license"></a>
+
+This project is [MIT](./MIT.md) licensed.
```
Comment (t-yanick) [inline README.md]: - [ ] When I click on the MIT license link, I am redirected to an error 404 page. Kindly fix this issue as it is not professional to keep broken links in our README.
image

File: README.md [hunk 3]
```diff
@@ -1,24 +1,216 @@
...
+<!-- PROJECT DESCRIPTION -->
 
-* Configuration
+# 📖 [Budget App] <a name="about-project"></a>
 
-* Database creation
+**[Buget App]** is a classic example of a blog website. It is a functional website that show the list of Categories and associated operations to each categories.
 
-* Database initialization
+**[Video Description](https://drive.google.com/file/d/1R-GI7cmCG04wLd3bvoDH8fmlJsp5wSMx/view?usp=drive_link)**
```
Comment (t-yanick) [inline README.md]: - [ ] You were recommended to use Loom for your video presentation. You could also use any other tool as long as you share an online link of the uploaded video. Unfortunately you have a google drive link but you have not granted access to everyone having the link. Kindly change the access rights.

File: README.md [hunk 4]
```diff
@@ -1,24 +1,216 @@
...
+- **[Add_New_Category]**
+- **[Add_New_Transaction]**
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- LIVE DEMO -->
+
+## 🚀 Live Demo <a name="live-demo"></a>
+
+[Live Demo Link](https://budgetapp-fnyd.onrender.com)
```
Comment (t-yanick) [inline README.md:85]: - [ ] As per project requirements, the deployed link should work as expected. At the moment, I am unable to SignUp. When I fill the form and click on Sign Up, nothing happens. Please check and ensure that the Live link is working.
image

**Review by t-yanick — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED

Hi @Outkarm , 

### Good job so far!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Use of Postgres for database and Devise for authentication
:heavy_check_mark: Tests added
:heavy_check_mark: Associations between models correctly setup
:heavy_check_mark: Necessary routes added
:heavy_check_mark: Good commit messages

There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 4 above)
Comment (Outkarm) [reply README.md:85]: just refresh the page

File: README.md [hunk 5]
```diff
@@ -1,24 +1,216 @@
...
+<!-- PROJECT DESCRIPTION -->
 
-* Configuration
+# 📖 [Budget App] <a name="about-project"></a>
 
-* Database creation
+**[Buget App]** is a classic example of a blog website. It is a functional website that show the list of Categories and associated operations to each categories.
 
-* Database initialization
+**[Video Description](https://drive.google.com/file/d/1R-GI7cmCG04wLd3bvoDH8fmlJsp5wSMx/view?usp=sharing)**
```
Comment (cynthiainga) [inline README.md]: - [ ] As the first reviewer was not able to open the video he was also not able to give feedback about it. So as mentioned in the project requirements you should highlight a piece of the code that you find interesting or are proud of making. Kindly record a new video that meets all the requirements 🙏🏾 

Screenshot 2023-06-04 at 00 11 08

File: README.md [hunk 6]
```diff
@@ -1,24 +1,216 @@
...
+- **[Add_New_Category]**
+- **[Add_New_Transaction]**
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- LIVE DEMO -->
+
+## 🚀 Live Demo <a name="live-demo"></a>
+
+[Live Demo Link](https://budgetapp-fnyd.onrender.com)
```
Comment (cynthiainga) [inline README.md:85]: - [ ] As mentioned by the previous reviewer, the deployed link should work properly, which means users don't need to refresh every time when they use your application. To create a category the button doesn’t work we need to refresh the page then it works, to add a transaction the button doesn’t work we need to refresh the page then it works after and that is a problem and it means your deployed link doesn’t work properly and it should work properly as requested. Your application should work without any problem without refreshing every time the page. Kindly fix that 🙃 🙏🏾

**Review by cynthiainga — CHANGES_REQUESTED** [review]
Hi @Outkarm, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.



 ## Highlights

- [ ] Used `devise` for authentication ✔️ 
- [ ] Used Postgres as the database for the application ✔️ 
- [ ] The app has the `Home` and `Transaction` pages ✔️ 

 ## Required Changes ♻️

- [ ] Please kindly check the comments under the review 👇🏾 👇🏾 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code. Some of them were missed by the previous reviewer and addressing them will really improve your application._


Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by zainsadaqat — APPROVED** [review]
# Pull Request Approved ✅ 🎊 🎉 
## Hello @Outkarm  👋 

### Your project is complete!  There is nothing else to say other than it's time to merge it.

# Congratulations!!! 🎉 🥳 👏 🙌 🎊 ☑️ 



Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**